### PR TITLE
drawContours: fix data types so that the function works even for huge images

### LINKED ISSUE #7458: drawContours fails for very large images
##### System information (version)
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2013
##### Detailed description

For very large images, the drawContours function does not produce the expected results.

See also #7449 – probably related. However, drawContours does not crash the program – it just produces an incorrect result.
##### Steps to reproduce

Please note that you'll need to have merged at least one of #7451 and #7452, because otherwise already the `findContours` call crashes the program:

```
cv::Mat largeImage(79925, 49955, CV_8UC1);
largeImage.setTo(0);

cv::circle(largeImage, cv::Point(largeImage.cols / 2, largeImage.rows / 2), 5000, 255, CV_FILLED);

std::vector<std::vector<cv::Point>> contours;
cv::findContours(largeImage, contours, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_SIMPLE);

largeImage.setTo(0);
cv::drawContours(largeImage, contours, -1, 255, CV_FILLED);

cv::Mat smallImage;
cv::resize(largeImage, smallImage, cv::Size(499, 799));

cv::imwrite("smallImage.png", smallImage);
```

Result (not ok):
smallimage

However, this code does work as expected (when pasted right after the above snippet, so that `contours` has already been found):

```
std::vector<std::vector<cv::Point>> contours2 = contours;
for (auto& contour : contours2) {
    for (auto& point : contour) {
        point.x /= 100;
        point.y /= 100;
    }
}

cv::Mat smallImage2(smallImage.size(), CV_8UC1);
smallImage2.setTo(0);
cv::drawContours(smallImage2, contours2, -1, 255, CV_FILLED);
cv::imwrite("smallImage2.png", smallImage2);
```

Result (ok):
smallimage2

I'll try to find and fix the problem myself.

### PR DESCRIPTION
**Pull request by reunanen**
resolves #7458

### CODE REVIEW

File: modules/core/src/drawing.cpp [hunk 1]
```diff
@@ -45,13 +45,23 @@ namespace cv
 
 enum { XY_SHIFT = 16, XY_ONE = 1 << XY_SHIFT, DRAWING_STORAGE_BLOCK = (1<<12) - 256 };
 
+static const int MAX_THICKNESS = 32767;
```
Comment (alalek) [inline modules/core/src/drawing.cpp]: Please remove this variable, it is not used.

Comment (reunanen) [inline modules/core/src/drawing.cpp]: Fixed.

**Comment by vpisarev** [comment]
@terfendail, can you please review the patch? I think, it relates to your other patch for drawing.cpp, even though that patch was for opencv 3.x. Shall we backport drawing.cpp to 2.4.x?

**Comment by alalek** [comment]
Replaced by #8031

**Pull request closed without merging**